fileghost/fileghost.py

```diff
@@ -1,114 +1,139 @@
-import secrets
 import json
-import sys
 import os
-global SEP
-SEP='__FIL3GH0ST__'
-global SEPARATOR
-SEPARATOR=list(bytes(SEP.encode()))
-
-def keystore_validity_test(keys:dict):
-    def valid_values():
-        flag=True
-        for v in keys.values():
-            if v>255:
-               flag=False
-               return flag
-               break
-        return flag
-    print("Checking for keystore validity...")
-    assert isinstance(keys,dict), "Keystore must be a dictionary"
-    assert len(keys)==256, "Keystore keys count must be 256, but "+str(len(keys))+" instead"
-    assert sum(keys.keys())==32640, "Sum of keystore keys should be 32640, but "+str(sum(keys.keys()))+" instead"
-    assert len(keys.values())==len(set(keys.values())), "Keystore values must be unique and must not repeat, but duplicates found"
-    assert valid_values()==True, "Any keystore values must be less than 256"
-    
-    return True
-class keygen():
-    def __init__(self):
-        global l
-        l=[]
-        while len(l) != 256:
-              r=secrets.randbelow(256)
-              if r not in l:
-                 l.append(r)
-    def to_keystore(self):#needed for file encryption/decryption
-        s = set(range(256))
-        keystore=dict.fromkeys(s)
-        for x in keystore:
-            keystore[x]=l[x]
-        return keystore
-        return hl
-
-    def to_byte_array(self):
-        return l    
+import secrets
+import sys
+
+
+SEP = '__FIL3GH0ST__'.encode()
+SEPARATOR = list(bytes(SEP))
+
+
+class keygen:
+    @classmethod
+    def generate(cls) -> "keygen":
+        '''
+        Generates a new keygen object with random keys, based on the ``secrets`` module.
+        '''
+        keys = []
+
+        numbers = set()
+        while len(numbers) < 256:
+            n = secrets.randbelow(256)
+
+            if n not in numbers:
+                numbers.add(n)
+                keys.append(n)
+
+        return cls(keys)
+
+    @classmethod
+    def from_file(cls, path: str) -> "keygen":
```

Comment:
Instead of having the user remember to read/write the keys (otherwise they can never decrypt their message/file), I added this `and `to_file` on the `keygen` class to make the process much easier